tools/format.py

```diff
@@ -26,7 +28,12 @@
 #   * These third party python files shouldn't be formatted.
 #   * The tools directory has no subdirectories, so `glob()` is sufficient.
 # TODO(ry) Install yapf in third_party.
-run(["yapf", "-i"] + glob("tools/*.py") + find_exts("build_extra", ".py"))
+if find_executable('yapf'):
+    run(["yapf", "-i"] + glob("tools/*.py") + find_exts("build_extra", ".py"))
+else:
+    print("\n\nUnable to format python code!: yapf not found."
+    "Install with \n`pip install yapf`\n exiting")
```

Comment:
It's surprising to me that yapf is happy  with this formatting.

I think it's cleaner to do:

```
     print("\nUnable to format python code: yapf not found.")
     print("Install with `pip install yapf`.")
```
Though ry may prefer old-style print statements...

Reply:
i can change